src/api/java/com/minecolonies/api/util/InventoryUtils.java

```diff
@@ -2084,6 +2084,40 @@ public static boolean removeStackFromItemHandler(final IItemHandler handler, fin
         return false;
     }
 
+    /**
+     * Remove a stack with a certain qty from a given Itemhandler
+     *
+     * @param handler the itemHandler.
+     * @param input   the stack to remove.
+     * @param count the amount to remove.
+     */
+    public static void removeStackFromItemHandler(final IItemHandler handler, final ItemStack input, final int count)
+    {
+        final ItemStack workingStack = input.copy();
+        int localCount = count;
+        int tries = 0;
+        while (tries < count)
+        {
+            final int slot = findFirstSlotInItemHandlerNotEmptyWith(handler, stack -> ItemStackUtils.compareItemStacksIgnoreStackSize(workingStack, stack));
+            if (slot == -1)
+            {
+                return;
+            }
+
+            final int removedSize = ItemStackUtils.getSize(handler.extractItem(slot, localCount, false));
```

Comment:
This does not work. You also need to check if removedSize is bigger then the localCount, so you will need to simulate first. You have no idea how the IItemHandler is implemented, if it even allows partial extraction. Just because you now only use it for extraction from the Player or Citizen does not mean this is a valid utility implementation, and we might possibly run into this problem in the future.